packages/cli/src/commands/setup/auth/auth.js

```diff
@@ -101,31 +109,128 @@ export async function builder(yargs) {
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-auth0-setup')
+        handler(args)
       }
-    }
+    )
+    .command(
+      'azure-active-directory',
+      'Generate an auth configuration for Azure Active Directory',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler(
+          '@redwoodjs/auth-azure-active-directory-setup'
+        )
+        handler(args)
+      }
+    )
+    .command(
+      'clerk',
+      'Generate an auth configuration for Clerk',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-clerk-setup')
+        handler(args)
+      }
+    )
+    .command(
+      'custom',
+      'Generate a custom auth configuration',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-custom-setup')
+        handler(args)
+      }
+    )
+    .command(
+      'dbauth',
+      'Generate an auth configuration for dbAuth',
+      (yargs) => {
+        return standardAuthBuilder(yargs).option('webauthn', {
+          alias: 'w',
+          default: null,
+          description: 'Include WebAuthn support (TouchID/FaceID)',
+          type: 'boolean',
+        })
+      },
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-dbauth-setup')
+        handler(args)
+      }
+    )
+    .command(
+      'firebase',
+      'Generate an auth configuration for Firebase',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-firebase-setup')
+        handler(args)
+      }
+    )
+    .command(
+      'netlify',
+      'Generate an auth configuration for Netlify',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-netlify-setup')
+        handler(args)
+      }
+    )
+    .command(
+      'supabase',
+      'Generate an auth configuration for Supabase',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler('@redwoodjs/auth-supabase-setup')
+        handler(args)
+      }
+    )
+    .command(
+      'supertokens',
+      'Generate an auth configuration for SuperTokens',
+      (yargs) => standardAuthBuilder(yargs),
+      async (args) => {
+        const handler = await getAuthHandler(
+          '@redwoodjs/auth-supertokens-setup'
+        )
+        handler(args)
+      }
+    )
+}
 
-    if (commandModule) {
-      setupAuthCommand.command(commandModule)
-    }
+async function getAuthHandler(module) {
+  const { version } = fs.readJSONSync(
+    path.resolve(__dirname, '../../../../package.json')
+  )
+
+  if (!isInstalled(module)) {
+    await execa.command(`yarn add -D ${module}@${version}`, {
```

Comment:
Ran into issues when I tested this in a local project that I had `project:sync`ed

![image](https://user-images.githubusercontent.com/30793/208959444-74634997-e1d4-4a93-96bb-fc65ba7b9464.png)

I'll try to come up with a solution

Reply:
I found a solution
I check npm to make sure it exists. If it doesn't I fallback to "canary".
@jtoar How do you like that solution? Feel free to hate it 😁

Reply:
@Tobbe I like your approach but I don't like that it involves adding another dependency to the CLI so I just pushed a commit that uses yarn instead to do the same thing I think. Let me know what you think

Reply:
I had a feeling you wouldn't like the extra package. I thought about `fetch` to query the npm api myself. But using yarn to do it for us like you did is pretty clever 🙂